### PR TITLE
[MAMAEdu-123073] - Archive and Online Courses tabs

### PR DESCRIPTION
**Pull request by Inferato**
**AC:** 

- The new tabs are shown: "Online Courses" and "Archive";
- Button "Show course" doesn`t appear for all users;
- "Move to archive" button must be shown instead of "Show course";
- "Move to archive" button shown only for admin and staff roles;
- Course is moving to the archive by pressing "Move to archive" button;
- Must be able to re-run course from the archive(decided to provide re-run as recover option with @GlugovGrGlib );
- By default "Online courses" tab is active;
- If no courses in "Online courses" - "Archive" are shown;

**Youtrack:** https://youtrack.raccoongang.com/issue/MAMAEdu-123073

**Reviewers:**
- [ ] @GlugovGrGlib 
- [ ] @dyudyunov 
- [ ] @ViktorRusakov 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: cms/templates/index.html [hunk 1]
```diff
@@ -372,10 +374,31 @@ <h3 class="course-title">${course_info['display_name']}</h3>
               </li>
               % endif
               <li class="action action-view">
-                <a href="${course_info['lms_link']}" rel="external" class="button view-button">${_("View Live")}</a>
+                <script>
+                  function changeArch(event) {
+                      var course_key = event.target.value
+                      console.log(course_key)
```
Comment (ViktorRusakov) [inline cms/templates/index.html:380]: Should probably remove this line

File: cms/templates/index.html [hunk 2]
```diff
@@ -338,16 +338,18 @@ <h3 class="course-title">${course_info['display_name']}</h3>
...
       </ul>
       % endif
 
-      %if len(courses) > 0 or optimization_enabled:
+      %if (len(courses) > 0 or optimization_enabled) and (len(courses) != len([course_info for course_info in courses if course_info['archived'] == True])):
       <div class="courses courses-tab active">
         <ul class="list-courses">
           %for course_info in sorted(courses, key=lambda s: s['display_name'].lower() if s['display_name'] is not None else ''):
-          <li class="course-item" data-course-key="${course_info['course_key']}">
+          % if course_info['archived'] == False:
```
Comment (ViktorRusakov) [inline cms/templates/index.html:351]: You don't need to use == here, it is not a good practice to compare booleans like that, it's better to just write 'if not course_info['archived']'

File: cms/templates/index.html [hunk 3]
```diff
@@ -495,6 +523,74 @@ <h4 class="title">${_('Your Course Creator Request Status:')}</h4>
       </div>
       % endif
 
+
+      %if len(courses) > 0 or optimization_enabled:
+      <div class="archive archive-tab">
+        <ul class="list-courses">
+          %for course_info in sorted(courses, key=lambda s: s['display_name'].lower() if s['display_name'] is not None else ''):
+          % if course_info['archived'] == True:
```
Comment (ViktorRusakov) [inline cms/templates/index.html:531]: same as above - no need for == here, you can just write 'if course_info['archived']'

File: cms/templates/index.html [hunk 4]
```diff
@@ -495,6 +523,74 @@ <h4 class="title">${_('Your Course Creator Request Status:')}</h4>
...
+                    <span class="value">${course_info['number']}</span>
+                  </span>
+                  <span class="course-run metadata-item">
+                    <span class="label">${_("Course Run:")}</span> <span class="value">${course_info['run']}</span>
+                  </span>
+                </div>
+              </a>
+
+              <ul  class="item-actions course-actions">
+                % if allow_course_reruns and rerun_creator_status and course_creator_status=='granted':
```
Comment (ViktorRusakov) [inline cms/templates/index.html:551]: please add spaces around == at the end of the line

Comment (Inferato) [reply cms/templates/index.html:551]: Sorry, didn't notice about this. I'm borrow this part from course card

**Review by dyudyunov — APPROVED** [review]

**Review by ViktorRusakov — APPROVED** [review]

**Pull request merged**